File: packages/jsii-pacmak/test/expected.jsii-calc-base/java/src/main/java/software/amazon/jsii/tests/calculator/base/$Module.java
package software.amazon.jsii.tests.calculator.base;

import static java.util.Arrays.asList;

import java.util.List;
import software.amazon.jsii.JsiiModule;

public final class $Module extends JsiiModule {
    public $Module() {
        super("@scope/jsii-calc-base", "0.19.0", $Module.class, "jsii-calc-base@0.19.0.jsii.tgz");
    }

    @Override
    public List<Class<? extends JsiiModule>> getDependencies() {
        return asList(software.amazon.jsii.tests.calculator.baseofbase.$Module.class);
    }

    @Override
    protected Class<?> resolveClass(final String fqn) throws ClassNotFoundException {
        switch (fqn) {
            case "@scope/jsii-calc-base.Base": return software.amazon.jsii.tests.calculator.base.Base.class;
            case "@scope/jsii-calc-base.BaseProps": return software.amazon.jsii.tests.calculator.base.BaseProps.class;
            case "@scope/jsii-calc-base.IBaseInterface": return software.amazon.jsii.tests.calculator.base.IBaseInterface.class;
            default: throw new ClassNotFoundException("Unknown JSII type: " + fqn);
        }
    }
}

File: packages/jsii-pacmak/test/expected.jsii-calc-base/kotlin/src/main/kotlin/software/amazon/jsii/tests/kotlin/calculator/base/$Module.kt
package software.amazon.jsii.tests.kotlin.calculator.base

class `$Module`: software.amazon.jsii.JsiiModule("@scope/jsii-calc-base", "0.0.0", `$Module`::class.java, "jsii-calc-base@0.0.0.jsii.tgz") {
    override fun resolveClass(fqn: String?): Class<*> {
        return when (fqn) {
            "@scope/jsii-calc-base.Base" -> software.amazon.jsii.tests.kotlin.calculator.base.Base::class.java
            "@scope/jsii-calc-base.BaseProps" -> software.amazon.jsii.tests.kotlin.calculator.base.BaseProps::class.java
            "@scope/jsii-calc-base.IBaseInterface" -> software.amazon.jsii.tests.kotlin.calculator.base.IBaseInterface::class.java
            else -> throw ClassNotFoundException("Unknown JSII type: $fqn")
        }
    }
}
